src/graphql/admin/root/query/account-details-by-username.ts

```diff
@@ -0,0 +1,26 @@
+import { GT } from "@graphql/index"
+
+import GraphQLAccount from "@graphql/admin/types/object/account"
+import Username from "@graphql/types/scalar/username"
+import { getAccountByUsername } from "@app/admin"
+
+const AccountDetailsByUsernameQuery = GT.Field({
+  type: GT.NonNull(GraphQLAccount),
+  args: {
+    username: { type: GT.NonNull(Username) },
+  },
+  resolve: async (parent, { username }) => {
+    if (username instanceof Error) {
+      throw username
```

Comment:
Same comment `return` vs. `throw`?

Reply:
in general queries return object doesn't include `errors` so, at least in queries, throw is necessary (although for app returned errors is necessary to do the mapping)

Reply:
but the same object are being used whether it's a `query` of a `mutation`? like in both case we can return an `Account` object for instance? for a query because we fetch account details, for a mutation because we update the account status.